### ISSUE TITLE
Bluescreen (KERNEL_SECURITY_CHECK_FAILURE) on wake/sleep

### ISSUE THREAD

**Issue by felixrieseberg** [electron/electron#11649]
We have received reports from multiple Windows users on a blue screen due to `KERNEL_SECURITY_CHECK_FAILURE` during wake, sleep, closing the lid, or other sleep-related operations. There is little actionable information, but I'm reporting this in case someone else is seeing something similar so that we can collect as much information as possible.

* Electron version: 1.8.3-beta.2
* Operating system: Windows, build 10.0.14393.xxx is most common.
* Hardware: We've seen this across makes and models.

### Expected behavior
No bluescreen

### Actual behavior
Bluescreen

### How to reproduce
This is sadly entirely unclear.

<details>

```
KERNEL_SECURITY_CHECK_FAILURE (139) 
A kernel component has corrupted a critical data structure. The corruption 
could potentially allow a malicious user to gain control of this machine. 
Arguments: 
Arg1: 0000000000000003, A LIST_ENTRY has been corrupted (i.e. double remove). 
Arg2: ffff830094e58e40, Address of the trap frame for the exception that caused the bugcheck 
Arg3: ffff830094e58d98, Address of the exception record for the exception that caused the bugcheck 
Arg4: 0000000000000000, Reserved

Debugging Details: 
------------------ 
DUMP_CLASS: 1 
DUMP_QUALIFIER: 400 
BUILD_VERSION_STRING: 10.0.14393.693 (rs1_release.161220-1747) 
SYSTEM_MANUFACTURER: LENOVO 
SYSTEM_PRODUCT_NAME: 20F9005AUS 
SYSTEM_SKU: LENOVO_MT_20F9_BU_Think_FM_ThinkPad T460s 
SYSTEM_VERSION: ThinkPad T460s 
BIOS_VENDOR: LENOVO 
BIOS_VERSION: N1CET48W (1.16 ) 
BIOS_DATE: 09/16/2016 
BASEBOARD_MANUFACTURER: LENOVO 
BASEBOARD_PRODUCT: 20F9005AUS 
BASEBOARD_VERSION: SDK0J40697 WIN 
DUMP_TYPE: 2 
BUGCHECK_P1: 3 
BUGCHECK_P2: ffff830094e58e40 
BUGCHECK_P3: ffff830094e58d98 
BUGCHECK_P4: 0 
TRAP_FRAME: ffff830094e58e40 -- (.trap 0xffff830094e58e40) 
NOTE: The trap frame does not contain all registers. 
Some register values may be zeroed or incorrect. 
rax=ffffaa09ed5f1890 rbx=0000000000000000 rcx=0000000000000003 
rdx=fffff802eb52db00 rsi=0000000000000000 rdi=0000000000000000 
rip=fffff802eb7d12e5 rsp=ffff830094e58fd0 rbp=ffff830094e59100 
r8=ffffaa09ed5f1848 r9=0000000000000003 r10=7fffaa09ed5f1848 
r11=7ffffffffffffffc r12=0000000000000000 r13=0000000000000000 
r14=0000000000000000 r15=0000000000000000 
iopl=0 nv up ei ng nz na pe cy 
nt! ?? ::NNGAKEGL::`string'+0x4a0a5: 
fffff802`eb7d12e5 cd29 int 29h 
Resetting default scope

EXCEPTION_RECORD: ffff830094e58d98 -- (.exr 0xffff830094e58d98) 
ExceptionAddress: fffff802eb7d12e5 (nt! ?? ::NNGAKEGL::`string'+0x000000000004a0a5) 
ExceptionCode: c0000409 (Security check failure or stack buffer overrun) 
ExceptionFlags: 00000001 
NumberParameters: 1 
Parameter[0]: 0000000000000003 
Subcode: 0x3 FAST_FAIL_CORRUPT_LIST_ENTRY

CPU_COUNT: 4 
CPU_MHZ: af8 
CPU_VENDOR: GenuineIntel 
CPU_FAMILY: 6 
CPU_MODEL: 4e 
CPU_STEPPING: 3 
CPU_MICROCODE: 6,4e,3,0 (F,M,S,R) SIG: 9E'00000000 (cache) 9E'00000000 (init) 
CUSTOMER_CRASH_COUNT: 1

DEFAULT_BUCKET_ID: LIST_ENTRY_CORRUPT 
BUGCHECK_STR: 0x139 
PROCESS_NAME: Slack.exe 
CURRENT_IRQL: 1

ERROR_CODE: (NTSTATUS) 0xc0000409 - The system detected an overrun of a stack-based buffer in this application. This overrun could potentially allow a malicious user to gain control of this application.

EXCEPTION_CODE: (NTSTATUS) 0xc0000409 - The system detected an overrun of a stack-based buffer in this application. This overrun could potentially allow a malicious user to gain control of this application.

EXCEPTION_CODE_STR: c0000409 
EXCEPTION_PARAMETER1: 0000000000000003

ANALYSIS_SESSION_HOST: SHOLOD-HDT460S 
ANALYSIS_SESSION_TIME: 02-06-2017 17:32:23.0467 
ANALYSIS_VERSION: 10.0.14321.1024 amd64fre

LAST_CONTROL_TRANSFER: from fffff802eb362829 to fffff802eb3576f0

STACK_TEXT: 
ffff8300`94e58b18 fffff802`eb362829 : 00000000`00000139 00000000`00000003 ffff8300`94e58e40 ffff8300`94e58d98 : nt!KeBugCheckEx 
ffff8300`94e58b20 fffff802`eb362b90 : 00000000`00000000 ffffffff`80001938 ffffaa09`f7129d84 00000000`00000000 : nt!KiBugCheckDispatch+0x69 
ffff8300`94e58c60 fffff802`eb361b73 : ffffaa09`e14ac3d0 00000000`5e302b08 ffff8300`94e59340 00000000`000031d0 : nt!KiFastFailDispatch+0xd0 
ffff8300`94e58e40 fffff802`eb7d12e5 : ffff8300`94e59100 ffffaa09`00000002 ffffaa09`ed5f1820 ffffaa09`ed5f1820 : nt!KiRaiseSecurityCheckFailure+0xf3 
ffff8300`94e58fd0 fffff802`eb6e69cb : 00000000`00000000 00000000`3537d7d1 ffff8300`94e59100 ffffaa09`ed5f1820 : nt! ?? ::NNGAKEGL::`string'+0x4a0a5 
ffff8300`94e59000 fffff802`eb68a4c2 : 00000000`00000000 ffff8300`94e59470 ffff8300`94e59401 00000000`00000000 : nt!CmpDoParseKey+0x2adb 
ffff8300`94e593d0 fffff802`eb6e02e1 : fffff802`eb68a1c0 fffff803`00000001 00000000`00000000 ffff8300`94e59801 : nt!CmpParseKey+0x302 
ffff8300`94e59570 fffff802`eb692cdd : ffff990e`ab6b1b01 ffff8300`94e597d0 00000000`00000040 ffff990e`9c108bb0 : nt!ObpLookupObjectName+0xb71 
ffff8300`94e59740 fffff802`eb6929bd : ffffaa0a`00000001 000000e9`24ffee20 00000000`00000001 00000000`00000000 : nt!ObOpenObjectByNameEx+0x1dd 
ffff8300`94e59880 fffff802`eb6902ff : 0000027e`03227001 00000000`00001400 00007fff`4b33df40 00007fff`4b395020 : nt!CmOpenKey+0x29d 
ffff8300`94e59a40 fffff802`eb362393 : ffff990e`a84687c0 ffff990e`00000000 00000000`00000000 00000000`00000000 : nt!NtOpenKeyEx+0xf 
ffff8300`94e59a80 00007fff`58e282e4 : 00000000`00000000 00000000`00000000 00000000`00000000 00000000`00000000 : nt!KiSystemServiceCopyEnd+0x13 
000000e9`24ffed48 00000000`00000000 : 00000000`00000000 00000000`00000000 00000000`00000000 00000000`00000000 : 0x00007fff`58e282e4

STACK_COMMAND: kb

THREAD_SHA1_HASH_MOD_FUNC: d4ebd809b295e74f12cd19fb6449617794cb2876

THREAD_SHA1_HASH_MOD_FUNC_OFFSET: 68a20eff5f5e5ec05b0a81f45e23932dccc5ef53

THREAD_SHA1_HASH_MOD: dc844b1b94baa204d070855e43bbbd27eee98b94

FOLLOWUP_IP: 
nt!KiFastFailDispatch+d0 
fffff802`eb362b90 c644242000 mov byte ptr [rsp+20h],0

FAULT_INSTR_CODE: 202444c6

SYMBOL_STACK_INDEX: 2 
SYMBOL_NAME: nt!KiFastFailDispatch+d0 
FOLLOWUP_NAME: MachineOwner 
MODULE_NAME: nt 
IMAGE_NAME: ntkrnlmp.exe 
DEBUG_FLR_IMAGE_TIMESTAMP: 585a2651 
IMAGE_VERSION: 10.0.14393.693 
BUCKET_ID_FUNC_OFFSET: d0 
FAILURE_BUCKET_ID: 0x139_3_nt!KiFastFailDispatch 
BUCKET_ID: 0x139_3_nt!KiFastFailDispatch 
PRIMARY_PROBLEM_CLASS: 0x139_3_nt!KiFastFailDispatch 
TARGET_TIME: 2017-02-06T22:06:43.000Z 
OSBUILD: 14393 
OSSERVICEPACK: 693 
SERVICEPACK_NUMBER: 0 
OS_REVISION: 0 
SUITE_MASK: 272 
PRODUCT_TYPE: 1 
OSPLATFORM_TYPE: x64 
OSNAME: Windows 10 
OSEDITION: Windows 10 WinNt TerminalServer SingleUserTS 
OS_LOCALE: 
USER_LCID: 0 
OSBUILD_TIMESTAMP: 2016-12-21 01:50:57 
BUILDDATESTAMP_STR: 161220-1747 
BUILDLAB_STR: rs1_release 
BUILDOSVER_STR: 10.0.14393.693 
ANALYSIS_SESSION_ELAPSED_TIME: 4db 
ANALYSIS_SOURCE: KM 
FAILURE_ID_HASH_STRING: km:0x139_3_nt!kifastfaildispatch 
FAILURE_ID_HASH: {36173680-6f08-995f-065a-3d368c996911}
```

</details>

**Comment by sofianguy** [comment]
@felixrieseberg Is this bug present in Electron v2.0.x or later?

**Comment by sofianguy** [comment]
@felixrieseberg we are not releasing bug fixes earlier than 2.0.x. Feel free to reopen if this is still an issue in versions later than 2.0.x
